scripts/src/public/services/orders.ts

```diff
@@ -550,3 +550,43 @@ export async function createOutgoingTransferOrder(recipientWalletAddress: string
 
 	return openOrderDbToApi(order, sender.id);
 }
+
+export async function createIncomingTransferOrder(title: string, description: string, memo: string, senderWalletAddress: string, appId: string, receiver: User, receiverDeviceId: string): Promise<Order> {
+	logger().info("creating an incoming transfer order");
+	const receiverWallet = (await receiver.getWallets(receiverDeviceId)).lastUsed();
+	if (!receiverWallet) {
+		throw UserHasNoWallet(receiver.id, receiverDeviceId);
+	}
+
+	const order = db.IncomingTransferOrder.new({
+		offerId: `cross-app_${appId}_to_${receiver.appId}`,
+		status: "pending",
+		amount: 0, // not allowed to insert an order without amount
+		blockchainData: {
+			sender_address: senderWalletAddress,
+			recipient_address: receiverWallet.address,
+			memo
+		}
+	}, {
+		user: receiver,
+		type: "earn",
+		wallet: receiverWallet.address,
+		meta: { title, description }
+	});
+
+	await order.save();
+
+	// we create a redis entry so this memo can uniquely identify this order.id down the line
+	// i.e. when the watch we gonna add triggers a callback
+	const parts = memo.split("-"); // extract order id from memo
+	const redis = getRedisClient();
+	await redis.async.set(`transfer:${parts[2]}`, order.id);
```

Comment:
Maybe extract these 3 lines to a utility function

Reply:
typescript ftw